pytensor/compile/function/types.py

```diff
@@ -979,26 +975,26 @@ def restore_defaults():
                 # old-style linkers raise their own exceptions
                 raise
 
-        dt_fn = time.perf_counter() - t0_fn
-        self.maker.mode.fn_time += dt_fn
         if profile:
+            dt_fn = time.perf_counter() - t0_fn
+            self.maker.mode.fn_time += dt_fn
             profile.vm_call_time += dt_fn
 
         # Retrieve the values that were computed
         if outputs is None:
             outputs = [x.data for x in self.output_storage]
-        assert len(outputs) == len(self.output_storage)
+        else:
+            assert len(outputs) == len(self.output_storage)
```

Comment:
Because by definition in the previous statement they match

Reply:
Sure, I guess it depends what meaning you put in the `assert`. I think I tend to see it as meaning "now this is true". In this vision it would make more sense to have cover both branches rather than having to check mentally that the other branch is also covered.
But it's a question of preference, it doesn't matter.

Reply:
Again note I'm trying to reduce the function overhead. Checking a list has the same length as the list comprehension it was defined from is silly in that regard

Reply:
I can just remove the assert instead